adafruit_miniqr.py

```diff
@@ -0,0 +1,544 @@
+#pylint: enable=line-too-long
+
+class QRCode:
+    """The generator class for QR code matrices"""
+    def __init__(self, *, qr_type=None, error_correct=L):
+        """Initialize an empty QR code. You can define the `qr_type` (size)
+        of the code matrix, or have the libary auto-select the smallest
+        match. Default `error_correct` is type L (7%), but you can select M,
+        Q or H."""
+        self.type = qr_type
+        self.ECC = error_correct  #pylint: disable=invalid-name
+        self.matrix = None
+        self.module_count = 0
+        self.data_cache = None
+        self.data_list = []
+
+    def add_data(self, data):
+        """Add more data to the QR code, must be bytestring stype"""
+        self.data_list.append(data)
+        datalen = sum([len(x) for x in self.data_list])
+        if not self.type:
+            for qr_type in range(1, 6):
+                rs_blocks = _get_rs_blocks(qr_type, self.ECC)
+                total_data_count = 0
+                for block in rs_blocks:
+                    total_data_count += block['data']
+                if total_data_count > datalen:
+                    self.type = qr_type
+                    break
+        self.data_cache = None
+
+    def make(self, *, test=False, mask_pattern=0):
+        """Perform the actual generation of the QR matrix. To keep things
+        small and speedy we don't generate all 8 mask patterns and pick
+        the best. Instead, please pass in a desired mask_pattern, the
+        default mask is 0."""
+        self.module_count = self.type * 4 + 17
+        self.matrix = QRBitMatrix(self.module_count, self.module_count)
+
+        self.setup_position_probe_pattern(0, 0)
+        self.setup_position_probe_pattern(self.module_count - 7, 0)
+        self.setup_position_probe_pattern(0, self.module_count - 7)
+        self.setup_position_adjust_pattern()
+        self.setup_timing_pattern()
+        self.setup_type_info(test, mask_pattern)
+
+        if self.type >= 7:
+            self.setup_type_number(test)
+
+        if self.data_cache is None:
+            self.data_cache = QRCode.create_data(self.type, self.ECC, self.data_list)
+        self.map_data(self.data_cache, mask_pattern)
+
+    def setup_position_probe_pattern(self, row, col):
```

Comment:
Should all of these start with _ to be private? This is an interesting API comparison: https://github.com/lincolnloop/python-qrcode

Reply:
i dont know if you'd ever wanna use them outside the library but i could do it sure - i ported this, which i think is a port of the above
https://github.com/SHA2017-badge/micropython-examples-sha2017/tree/master/QRcode

Reply:
Do you want to change it now or wait? I was thinking its a good change because it reduces the public API complexity.

Reply:
yeah ill do it now one minute